tests/test_endpoint.py

```
@@ -137,6 +233,15 @@ def test_quic_endpoint_rejects_an_active_generated_connection_id(initial: bytes)
         endpoint.receive_datagram(another.data_to_send()[0], b"address", 0)
 
 
+def test_quic_endpoint_requires_endpoint_connection_id_issuance(initial: bytes) -> None:
+    endpoint = zttp.QuicEndpoint(connection_id_factory=lambda length: b"s" * length)
+    connection = endpoint.receive_datagram(initial, b"address", 0)
+    assert connection is not None
+
+    with pytest.raises(zttp.LocalProtocolError, match="QuicEndpoint.issue_connection_id"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)